### DISCUSSION TITLE
Show your noble projects

### DISCUSSION THREAD

**Discussion by paulmillr** [paulmillr/noble-curves#90] [Show and tell]
Noble cryptography is used by protonmail, metamask, rainbow, nostr and others.

Does your project also use noble? Post it here! I will add it to the website https://paulmillr.com/noble/ and you will get new users.

**Comment by mahnunchik** [comment]
Version 6.0.0 of [Coin Wallet](https://github.com/CoinSpace/CoinSpace) has been rewritten with extensive use of the `@scure/*` and `@noble/*` modules.

* https://github.com/CoinSpace/cs-bitcoin-wallet - build with `@noble/curves`, `@noble/hashes`, `@scure/base`, `@scure/bip32`, and **`@scure/btc-signer`**. It would be nice to remove last dependency when [CashAddr](https://github.com/paulmillr/scure-base/pull/24) will be supported.
* https://github.com/CoinSpace/cs-evm-wallet - `@noble/hashes`, `@scure/bip32`, and **`micro-eth-signer`**.
* https://github.com/CoinSpace/cs-solana-wallet - `@noble/curves`, `@scure/base`, `ed25519-keygen`, and **`micro-sol-signer`**.
* https://github.com/CoinSpace/cs-eos-wallet - implemented on top of `@noble/curves`, `@noble/hashes`, and `@scure/base` only.
* https://github.com/CoinSpace/cs-tron-wallet - `@noble/hashes`, `@scure/base`, `@scure/bip32`, and `@noble/curves` through `tronlib`.
* https://github.com/CoinSpace/cs-monero-wallet - `@scure/bip32`, `@noble/hashes`, `@scure/base`.
* https://github.com/CoinSpace/cs-cardano-wallet - `@noble/hashes` and `@scure/base`.

**Reply by paulmillr** [reply]
That's awesome!!! Will add your app to noble homepage.

**Comment by mahnunchik** [comment]
@paulmillr Congrats with adoption of `@noble/*` and `@scure/*` libraries in `xrpl.js` codebase! 🎉

https://github.com/XRPLF/xrpl.js/releases/tag/xrpl%403.0.0

I think this will reduce the size of Coin Wallet by several KB!

My own wish list for next:
* `monero` - it is pain
* `stellar`
* `cardano`

**Reply by paulmillr** [reply]
Great stuff!

**Comment by imcotton** [comment]
I made a thin wrapper on top of `ed25519-keygen/ssh` via **Deno** and **JSR**[^jsr], to help user to _**delete**_ their SSH private keys, just submit to Show Hacker News[^hn], credit been given at the end of the post.

[^jsr]: https://jsr.io/@key/gen-ssh-ed25519
[^hn]: https://news.ycombinator.com/item?id=39684380

**Reply by paulmillr** [reply]
I like it! It’s great!

One thing that can be done is replacing pbkdf2 with scrypt.

**Reply by imcotton** [reply]
Thanks, yes **scrypt** or even **Argon2** would be better choose over **PBKDF2**, I picked it for the  sake of **FIPS** compliance, also offload functionality into native **Web Crypto**, might revisit it in future by major version bumps.

**Comment by pete-watters** [comment]
Hi @paulmillr :wave:

We use a lot of your libraries in [Leather wallet](https://github.com/leather-io/extension) - https://leather.io/ and are very appreciative of your great work. 

Can you please add Leather wallet under Wallets on your website?

**Reply by paulmillr** [reply]
Added!

**Reply by pete-watters** [reply]
Thanks! 🙇

**Comment by li0ard** [comment]
Hi, I recently ported my library from libsodium to `@noble/*`. The library is designed to work with user ID in decentralized messenger [Session](https://getsession.org) - [session_id](https://github.com/theinfinityway/session_id)

**Reply by paulmillr** [reply]
Nice. Added to the list.

**Reply by li0ard** [reply]
It also appears that [Alby wallet](https://getalby.com) (wallet for Nostr) also uses `@noble/*` and `@scure/*` libraries. example [#1](https://github.com/getAlby/lightning-browser-extension/blob/master/package.json#L42), [#2](https://github.com/getAlby/nostr-signer/blob/master/package.json#L19)

**Comment by dillonstreator** [comment]
Hi @paulmillr - I’m working on the project https://crypt.fyi, an open-source end to end encrypted ephemeral secrets-sharing platform designed for secure, quick, and temporary sharing of sensitive information.

It uses `@noble/ciphers` and `@noble/hashes`, primarily because of the simple API design, cross-platform compatibility, and strong focus on security. These qualities make it an excellent fit for the zero-knowledge architecture and help ensure that secrets are handled with the highest level of protection.

Huge thanks for creating such well-crafted tools.

**Reply by paulmillr** [reply]
Looks great. Your pbkdf2 iteration count is too small, should be 2**19.

Also I assume "read count" tracking is done in the server?

**Reply by dillonstreator** [reply]
Thank you for the review and pointer - I will adjust
Yup, read count is handled atomically on the server

**Comment by li0ard** [comment]
👋 Hello again, @paulmillr! I am working on implementations of GOST cryptography on TS. Im using `@noble/hashes` as HMAC and PBKDF2 wrappers for hash functions [`GOST R 34.11-94`](https://github.com/li0ard/gost341194) and [`GOST R 34.11-2012`](https://github.com/li0ard/streebog) aka `Streebog`.

Part of your code (`@noble/hashes` and `@noble/curves`) is also used in implementations of the [`Kuznyechik`](https://github.com/li0ard/kuznyechik) (aka `GOST R 34.12-2015`) and [`Magma`](https://github.com/li0ard/magma) (aka `GOST R 34.12-2015` and `GOST 28147-89`) ciphers, specifically in implementations of encryption modes.

**Reply by paulmillr** [reply]
Noted.

I've heard the algorithms were backdoored: https://eprint.iacr.org/2016/071.pdf, https://who.paris.inria.fr/Leo.Perrin/pi.html

**Reply by li0ard** [reply]
As far as i know, it's about S-Box generation for Kuznyechik and that there is some structure in it. Official reason from developers: S-Box generation algorithm was lost in 2007 during algorithm development. Nevertheless, these algorithms are standardized and used in Russia. And at the moment it's still unknown whether backdoor is present in Kuznyechik and Streebog.

Magma is an analog of DES from the USSR (GOST 28147-89), and its revised version (GOST R 34.12-2015).

**Comment by axew3** [comment]
w3myPQ. A Post Quantum file sharing plugin for WordPress.
https://wordpress.org/plugins/w3mypq/

**Reply by axew3** [reply]
You may have never seen this done so easily 😎
Folder encryption will be added in next versions of both BPQ (BennyPQ) and w3myPQ.

https://github.com/axew3/BennyPQ

**Reply by axew3** [reply]
Hello again, i will not stress you more so often, just to let you know that
BPQ has been transformed into a Standalone Client Integration.
The How to link into the github repo page https://github.com/axew3/BennyPQ 
explain what w3myPQ and BPQ can do together https://www.axew3.com/w3/2026/09/how-to-decentralized-file-sharing-in-wordpress/
I really think (but maybe i am wrong) nothing else on the web compares to this level of absolute security. Nothing, if we assume for example, a fresh Debian install with a vanilla firefox! ahaha
I will github w3myPQ soon because i really would like to find out some help on improve it or break it, but this last i assume it is impossible!
